apps/checker/app/matchers/DictionaryMatcher.scala

```diff
@@ -0,0 +1,46 @@
+package matchers
+
+import com.gu.typerighter.model.{Category, DictionaryRule}
+import org.languagetool.{JLanguageTool, Language, ResultCache, UserConfig}
+import services.collins.{CollinsEnglish, MorfologikCollinsSpellerRule, SpellDictionaryBuilder}
+import services.MatcherRequest
+import utils.Matcher
+
+import scala.concurrent.ExecutionContext
+import scala.jdk.CollectionConverters.ListHasAsScala
+
+class DictionaryMatcher(
+    rules: List[DictionaryRule]
+) extends Matcher {
+  val language: Language = new CollinsEnglish()
+  val cache: ResultCache = new ResultCache(10000)
+  val userConfig: UserConfig = new UserConfig()
+
+  val instance = new JLanguageTool(language, cache, userConfig)
+  // As a side effect, make sure the .dict artefact is available to languageTool
+  new SpellDictionaryBuilder().buildDictionary(rules.map(rule => rule.word))
+
+  instance
+    .getAllRules()
+    .forEach(rule => {
+      if (rule.getId() != MorfologikCollinsSpellerRule.RULE_ID) {
+        instance.disableRule(rule.getId())
+      }
```

Comment:
Maybe adding a comment explaining why we're disabling rules here would be helpful? I'm a little confused because clicking through to `isDictionaryBasedSpellingRule` it looks like this method will always return `false`?

Reply:
You may be on an older version of the branch as I don't think it's using `isDictionaryBasedSpellingRule` any more - instead it uses the `MorfologikCollinsSpellerRule.RULE_ID` to identify the rules we want to keep.

I've added a comment anyway!